src/%IPM/Main.cls

```diff
@@ -2383,11 +2412,229 @@ ClassMethod UpdateLanguageExtensions(pVerbose As %Boolean = 0, pTestOnly As %Boo
+				Set useLocal = 0
+			}
+		}
+	}
+
+	// Now we know which exact version to install and we know whether we should install it with local artifact or get from registry,
+	// 3. Figure out which namespaces to install
+
+	// TODO: Let user interactively choose which namespaces to install IPM
+	Set baseNS = $Namespace
+	Set sc = ##class(%SYS.Namespace).ListAll(.allNS)
+	$$$ThrowOnError(sc)
+	// multi-dim array to store all non-%SYS explicit namepaces that 
+	// either have no IPM installed (if allow-upgrade is not set)
+	// or all of them (if allow-upgrade is set)
+	Kill namespacesForIPM
+	Set currentNS = ""
+	If allowUpgrade {
+		Merge namespacesForIPM = allNS
+	} Else {
+		While 1 {
+			Set currentNS = $ORDER(allNS(currentNS))
+			If currentNS = "" {
+				Quit
+			}
+			// avoid %SYS and implied namespaces
+			If (currentNS '= "%SYS") && ($ZCONVERT(currentNS, "U") = currentNS) {
+				New $Namespace
+				Set $Namespace = currentNS
+				If '$System.CLS.IsMthd("%IPM.Main", "Shell") {
+					Set namespacesForIPM(currentNS) = 1
+				}
+			}
+		}
+	}
+	Set $Namespace = baseNS
+	If globally {
+		Merge targetNamespaces = namespacesForIPM
+	} ElseIf (namespaces '= "") {
+		Set specifiedNSs = $ListFromString(namespaces, ",")
+		Set ptr = 0
+   		While $ListNext(specifiedNSs, ptr, value) {
+			If '$DATA(namespacesForIPM(value)) {
+				Write !, value_" is not a valid namespace to install IPM"_$Select(allowUpgrade: ". ", 1: ", or it already has IPM installed. ")_"Skipped."
+			} Else {
+				Set targetNamespaces(value) = 1
+			}
+		}
+	} Else {
+		// TODO: Implement this part
+		Write !, "Currently do not support interactively choose namespaces to install IPM. Use -namespaces or -globally modifier."
+		Return
+	}
+
+	// 4. Now that we got all the namespaces that need to install IPM, do the actual installation
+	If 'useLocal {
+		Set server = ##class(%IPM.Repo.Remote.Definition).DeploymentServerOpen(1,,.sc)
+		$$$ThrowOnError(sc)
+		If $IsObject(server) {
+			Set packageService = server.GetPackageService()
+			Set ipmRef = ##class(%IPM.Storage.ModuleInfo).%New()
+			// TODO: Make this ipm eventually once the package name is changed
+			Set ipmRef.Name = $$$IPMModuleName
+			If (targetVersion = "latest") {
+				// convert latest to semantic version so that it can be undestood by ModuleInfo class
+				Set server = ##class(%IPM.Repo.Remote.Definition).DeploymentServerOpen(1,,.sc)
+				$$$ThrowOnError(sc)
+				If $IsObject(server) {
+					Set targetVersion = server.GetPackageService().GetLatestIPMVersion()
+				}
+			}
+			Set ipmRef.VersionString = targetVersion
+			
+			// Retrieve module installer manifest
+			Set manifest = packageService.GetModuleInstallerManifest(ipmRef)
+			Set currentNS = ""
+			While 1 {
+				Set currentNS = $ORDER(targetNamespaces(currentNS))
+				If currentNS = "" {
+					Quit
+				}
+				New $Namespace
+				Set $Namespace = currentNS
+				Do $SYSTEM.OBJ.LoadStream(manifest, "ck/displayerror=0/displaylog=1")
```

Comment:
When loading, should only displayerror and not displaylog. Should use same flags for LoadStream here and Load below for local version so perhaps store flags in a local variable.